server/src/main/java/org/opensearch/index/engine/NRTReplicationEngine.java

```diff
@@ -143,7 +145,11 @@ public synchronized void updateSegments(final SegmentInfos infos) throws IOExcep
             // Commit and roll the translog when we receive a different generation than what was last received.
             // lower/higher gens are possible from a new primary that was just elected.
             if (incomingGeneration != lastReceivedGen) {
-                commitSegmentInfos();
+                if (shouldCommit) {
+                    commitSegmentInfos();
+                } else {
+                    this.lastCommittedSegmentInfos = store.readLastCommittedSegmentsInfo();
```

Comment:
I think we also need to sync transaction logs (we do today for node-node) on commits to make them durable ?
```
translogManager.syncTranslog
```